lib/hello_web/router.ex

```diff
@@ -18,6 +19,38 @@ defmodule HelloWeb.Router do
     pipe_through :browser
 
     get "/", PageController, :index
+    get "/hello", HelloController, :index
+    #get "/hello/:messenger", HelloController, :show
+    get "/hello/:username", HelloController, :show
+    # redirección
+    get "/redirect_test", PageController, :redirect_test
+    resources "/users", UserController
+    resources "/posts", PostController, only: [:index, :show]
+    resources "/comments", CommentController, except: [:delete]
+    resources "/reviews", ReviewController
+
+    resources "/users", UserController do
+      resources "/posts", PostController
+    end
+  end
+
+  scope "/admin", HelloWeb.Admin, as: :admin do
+    pipe_through :browser
+
+    resources "/images", ImageController
+    resources "/reviews", ReviewController
+    resources "/users", UserController
+  end
+
+
+  scope "/api", HelloWeb.Api, as: :api do
+    pipe_through :api
+
+    scope "/v1", V1, as: :v1 do
+      resources "/images",  ImageController
```

Comment:
@YanethM what a `resources` do?

Reply:
I associate it with the laravel framework that has a type of resource controller, although in phoenix they call it macro, which takes all HTTP requests and can handle only and except options.